spec.md

```diff
@@ -19,15 +19,242 @@
 
 ### Space Data Model
 
+`Space` properties:
+
+* `id` - deterministically set by server if not provided.
+* `type` - a sorted array of strings, MUST include the type `Space`.
+* `name` (optional)
+* `controller` - determined by server based on who creates the space originally
+
 #### Space JSON Representation
 
 #### Space CBOR Representation
 
-### Space Operations
+### Create Space operation
+
+To create a Space:
+
+* Perform an authenticated Create Space operation. The controller/invoker DID
+  of the operation is set as the Space's `controller`
+
+* If an `id` is provided in the Create Space request, it must start with `urn:uuid`.
+  If no `id` is provided, it will be generated by the storage server.
+
+* (Optional, out of scope) A given storage provider MAY impose additional
+  requirements in order to create a Space for a given controller, such as:
+  - a Verifiable Credential representing a pre-arranged onboarding coupon
+  - a proof of payment
+  - a proof of membership in an organization
+
+#### (HTTP API) POST `/spaces/`
+
+To create a space via HTTP API:
+
+```http
+POST /spaces/ HTTP/1.1
+Host: example.com
+Accept: application/json
+Content-type: application/json
+Authorization: ...
+
+{
+  "name": "Example space #1"
+}
+```
+
+Example success response:
+
+```http
+HTTP/1.1 201 Created
+Content-type: application/json
+Location: https://example.com/spaces/81246131-69a4-45ab-9bff-9c946b59cf2e
+
+{
+  "id": "81246131-69a4-45ab-9bff-9c946b59cf2e",
+  "type": ["Space"],
+  "name": "Example space #1",
+  "controller": "did:key:z6MkpBMbMaRSv5nsgifRAwEKvHHoiKDMhiAHShTFNmkJNdVW"
+}
+```
+
+Example error response (missing authorization, unable to determine controller):
+
+Example error response (invalid authorization):
+
+Example error response (missing or insufficient onboarding material provided):
+
+Example error response (invalid `id` provided):
+
+Example error response (a space with the specified `id` already exists):
+
+#### (Javascript API) Space .create()
+
+### Read Space operation
+
+* Requires appropriate authorization (root zcap invoked by the space's controller,
+  or a zcap granting permission to read a particular space)
+* Returns the details for the specified space `id`
+
+#### (HTTP API) GET `/spaces/{space_id}`
+
+Example request:
+
+```http
+GET /spaces/81246131-69a4-45ab-9bff-9c946b59cf2e HTTP/1.1
+Host: example.com
+Accept: application/json
+Authorization: ...
```

Comment:
```suggestion
Authorization: Signature keyId="did:key:z6MkpBMbMaRSv5nsgifRAwEKvHHoiKDMhiAHShTFNmkJNdVW#z6MkpBMbMaRSv5nsgifRAwEKvHHoiKDMhiAHShTFNmkJNdVW" ...
```

Up to you whether you want this to ugly up the example, but imho it can help illustrate the auth

Reply:
oh absolutely, I figured we'll come back and add the actual headers